# Análisis EDA y Decisiones de Negocio en Ventas de Automóviles

Este notebook contiene 15 ejercicios que combinan análisis exploratorio de datos (EDA) y reflexiones críticas de negocio. Cada punto incluye instrucciones detalladas, paso a paso, para realizar agrupaciones y filtros, de modo que se guíe al estudiante en cada parte del proceso.

La estructura del dataset es la siguiente:

```
year, make, model, trim, body, transmission, vin, state, condition, odometer, color, interior, seller, mmr, sellingprice, saledate
```

¡Comienza el análisis y sigue cada paso para proponer estrategias de negocio basadas en los resultados!

In [ ]:
# Descargar y descomprimir el dataset
!wget https://github.com/javierherrera1996/lecture_analytics/raw/main/cars_dataset.zip
!unzip -o cars_dataset.zip

--2025-03-13 21:33:39--  https://github.com/javierherrera1996/lecture_analytics/raw/main/cars_dataset.zip
Resolving github.com (github.com)... 140.82.113.4
Connecting to github.com (github.com)|140.82.113.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/lecture_analytics/main/cars_dataset.zip [following]
--2025-03-13 21:33:39--  https://raw.githubusercontent.com/javierherrera1996/lecture_analytics/main/cars_dataset.zip
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 19753181 (19M) [application/zip]
Saving to: ‘cars_dataset.zip.2’

cars_dataset.zip.2  100%[===================>]  18.84M  --.-KB/s    in 0.06s   

2025-03-13 21:33:39 (300 MB/s) - ‘cars_dataset.zip.2’ saved [19753

## 1. Dimensiones y Representatividad del Dataset

a) Carga el dataset (desde un archivo CSV) y muestra sus dimensiones (número de filas y columnas).

b) Describe brevemente qué representa cada fila y reflexiona: ¿Crees que el dataset es representativo del mercado de ventas de automóviles? Justifica tu respuesta.

In [ ]:
import pandas as pd

data = pd.read_csv('car_prices.csv')


In [ ]:
data

year    make                model         trim       body  \
0       2015     Kia              Sorento           LX        SUV   
1       2015     Kia              Sorento           LX        SUV   
2       2014     BMW             3 Series   328i SULEV      Sedan   
3       2015   Volvo                  S60           T5      Sedan   
4       2014     BMW  6 Series Gran Coupe         650i      Sedan   
...      ...     ...                  ...          ...        ...   
558832  2015     Kia                 K900       Luxury      Sedan   
558833  2012     Ram                 2500  Power Wagon   Crew Cab   
558834  2012     BMW                   X5    xDrive35d        SUV   
558835  2015  Nissan               Altima        2.5 S      sedan   
558836  2014    Ford                F-150          XLT  SuperCrew   

       transmission                vin state  condition  odometer   color  \
0         automatic  5xyktca69fg566472    ca        5.0   16639.0   white   
1         automatic  5xyktca69fg561319    ca        5.0    9393.0   white   
2         automatic  wba3c1c51ek116351    ca       45.0    1331.0    gray   
3         automatic  yv1612tb4f1310987    ca       41.0   14282.0   white   
4         automatic  wba6b2c57ed129731    ca       43.0    2641.0    gray   
...             ...                ...   ...        ...       ...     ...   
558832          NaN  knalw4d4xf6019304    in       45.0   18255.0  silver   
558833    automatic  3c6td5et6cg112407    wa        5.0   54393.0   white   
558834    automatic  5uxzw0c58cl668465    ca       48.0   50561.0   black   
558835    automatic  1n4al3ap0fc216050    ga       38.0   16658.0   white   
558836    automatic  1ftfw1et2eke87277    ca       34.0   15008.0    gray   

       interior                                             seller      mmr  \
0         black                            kia motors america  inc  20500.0   
1         beige                            kia motors america  inc  20800.0   
2         black             financial services remarketing (lease)  31900.0   
3         black                            volvo na rep/world omni  27500.0   
4         black             financial services remarketing (lease)  66000.0   
...         ...                                                ...      ...   
558832    black                                   avis corporation  35300.0   
558833    black                                    i -5 uhlmann rv  30200.0   
558834    black             financial services remarketing (lease)  29800.0   
558835    black  enterprise vehicle exchange / tra / rental / t...  15100.0   
558836     gray                   ford motor credit company llc pd  29600.0   

        sellingprice                                 saledate  
0            21500.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  
1            21500.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  
2            30000.0  Thu Jan 15 2015 04:30:00 GMT-0800 (PST)  
3            27750.0  Thu Jan 29 2015 04:30:00 GMT-0800 (PST)  
4            67000.0  Thu Dec 18 2014 12:30:00 GMT-0800 (PST)  
...              ...                                      ...  
558832       33000.0  Thu Jul 09 2015 07:00:00 GMT-0700 (PDT)  
558833       30800.0  Wed Jul 08 2015 09:30:00 GMT-0700 (PDT)  
558834       34000.0  Wed Jul 08 2015 09:30:00 GMT-0700 (PDT)  
558835       11100.0  Thu Jul 09 2015 06:45:00 GMT-0700 (PDT)  
558836       26700.0  Thu May 28 2015 05:30:00 GMT-0700 (PDT)  

[558837 rows x 16 columns]

In [ ]:
pd.DataFrame(data).shape

(558837, 16)

Sí, el DataFrame ofrece información valiosa y representativa para el mercado sobre las ventas de los vehículos en diferentes lugares, pero esta información no le serviria a otros lugares que no fueran los mencionados en la base, ya que no hay información general, si no específica para estos lugares.

## 2. Ventas Totales y Promedio

a) Calcula la suma total y el promedio de la columna `sellingprice`.

b) Caso de negocio: Si el precio promedio es bajo, ¿qué estrategias de precio o promociones sugerirías para aumentar la rentabilidad? Explica tu respuesta.

In [ ]:
sellingprice_sum = data['sellingprice'].sum()
sellingprice_mean = data['sellingprice'].mean()

In [ ]:
sellingprice_sum

7606367587.0

In [ ]:
sellingprice_mean

13611.358810003132

Sí el precio promedio es bajo, las estratégias de precio o de promociones que le sugeriría para aumentar su rentabilidad sería darle un valor agregado a cada vehículo, ya que esto llamaría la atención de los clientes, campañas de marketing donde promocionaríamos los vehículos por medio de estratégias como la gama, el precio y algunas promociones y garantias que le daríamos al consumidor.

## 3. Distribución de Ventas por Marca

a) Utiliza `value_counts()` para determinar cuántas ventas se registraron por cada marca (`make`).

b) Caso de negocio: ¿Qué marcas parecen dominar el mercado? Basado en esta distribución, ¿en qué marcas invertirías en publicidad y por qué?

In [ ]:
data['make'].value_counts()

make
Ford          93554
Chevrolet     60197
Nissan        53946
Toyota        39871
Dodge         30710
              ...  
airstream         1
ford tk           1
chev truck        1
hyundai tk        1
Lotus             1
Name: count, Length: 96, dtype: int64

En este mercado predominan las marcas: Ford, Chevrolet, Nissan, Toyota y Dodge, e invertiría en estratégias de publicidad en Ford y Chevrolet ya que tienen un volúmen de ventas mayor en el mercado y una mayor preferencia por parte de los clientes, lo que genera una mayor rentabilidad, y si implemento estratégias de publicidad esta rentabilidad podría aumentar.

## 4. Segmentación de Clientes por Kilometraje

a) Crea un DataFrame filtrado que incluya únicamente las ventas donde el `odometer` es menor a 20,000 millas. Cuantos Vehiculos tiene este dataset?

b) Caso de negocio: Este segmento de vehículos con bajo uso puede tener mayor valor percibido. ¿Cómo ajustarías la estrategia de marketing para este grupo?

In [ ]:
data_filtered = data[data['odometer'] < 20000]

In [ ]:
data_filtered

year    make                model        trim       body transmission  \
0       2015     Kia              Sorento          LX        SUV    automatic   
1       2015     Kia              Sorento          LX        SUV    automatic   
2       2014     BMW             3 Series  328i SULEV      Sedan    automatic   
3       2015   Volvo                  S60          T5      Sedan    automatic   
4       2014     BMW  6 Series Gran Coupe        650i      Sedan    automatic   
...      ...     ...                  ...         ...        ...          ...   
558812  2012  Nissan               Maxima      3.5 SV      sedan          NaN   
558826  2014    Jeep       Grand Cherokee     Limited        SUV    automatic   
558832  2015     Kia                 K900      Luxury      Sedan          NaN   
558835  2015  Nissan               Altima       2.5 S      sedan    automatic   
558836  2014    Ford                F-150         XLT  SuperCrew    automatic   

                      vin state  condition  odometer   color interior  \
0       5xyktca69fg566472    ca        5.0   16639.0   white    black   
1       5xyktca69fg561319    ca        5.0    9393.0   white    beige   
2       wba3c1c51ek116351    ca       45.0    1331.0    gray    black   
3       yv1612tb4f1310987    ca       41.0   14282.0   white    black   
4       wba6b2c57ed129731    ca       43.0    2641.0    gray    black   
...                   ...   ...        ...       ...     ...      ...   
558812  1n4aa5ap7cc840019    pa       42.0    9861.0   brown      tan   
558826  1c4rjebg4ec573100    ca       44.0    9024.0    gray    black   
558832  knalw4d4xf6019304    in       45.0   18255.0  silver    black   
558835  1n4al3ap0fc216050    ga       38.0   16658.0   white    black   
558836  1ftfw1et2eke87277    ca       34.0   15008.0    gray     gray   

                                                   seller      mmr  \
0                                 kia motors america  inc  20500.0   
1                                 kia motors america  inc  20800.0   
2                  financial services remarketing (lease)  31900.0   
3                                 volvo na rep/world omni  27500.0   
4                  financial services remarketing (lease)  66000.0   
...                                                   ...      ...   
558812                                 nissan-infiniti lt  18650.0   
558826  enterprise vehicle exchange / tra / rental / t...  29800.0   
558832                                   avis corporation  35300.0   
558835  enterprise vehicle exchange / tra / rental / t...  15100.0   
558836                   ford motor credit company llc pd  29600.0   

        sellingprice                                 saledate  
0            21500.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  
1            21500.0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  
2            30000.0  Thu Jan 15 2015 04:30:00 GMT-0800 (PST)  
3            27750.0  Thu Jan 29 2015 04:30:00 GMT-0800 (PST)  
4            67000.0  Thu Dec 18 2014 12:30:00 GMT-0800 (PST)  
...              ...                                      ...  
558812       19700.0  Tue Jul 07 2015 07:30:00 GMT-0700 (PDT)  
558826       17300.0  Thu Jul 09 2015 09:00:00 GMT-0700 (PDT)  
558832       33000.0  Thu Jul 09 2015 07:00:00 GMT-0700 (PDT)  
558835       11100.0  Thu Jul 09 2015 06:45:00 GMT-0700 (PDT)  
558836       26700.0  Thu May 28 2015 05:30:00 GMT-0700 (PDT)  

[84206 rows x 16 columns]

In [ ]:
pd.DataFrame(data_filtered).shape

(84206, 16)

In [ ]:
data.head()

year   make                model        trim   body transmission  \
0  2015    Kia              Sorento          LX    SUV    automatic   
1  2015    Kia              Sorento          LX    SUV    automatic   
2  2014    BMW             3 Series  328i SULEV  Sedan    automatic   
3  2015  Volvo                  S60          T5  Sedan    automatic   
4  2014    BMW  6 Series Gran Coupe        650i  Sedan    automatic   

                 vin state  condition  odometer  color interior  \
0  5xyktca69fg566472    ca        5.0   16639.0  white    black   
1  5xyktca69fg561319    ca        5.0    9393.0  white    beige   
2  wba3c1c51ek116351    ca       45.0    1331.0   gray    black   
3  yv1612tb4f1310987    ca       41.0   14282.0  white    black   
4  wba6b2c57ed129731    ca       43.0    2641.0   gray    black   

                                   seller      mmr  sellingprice  \
0                 kia motors america  inc  20500.0       21500.0   
1                 kia motors america  inc  20800.0       21500.0   
2  financial services remarketing (lease)  31900.0       30000.0   
3                 volvo na rep/world omni  27500.0       27750.0   
4  financial services remarketing (lease)  66000.0       67000.0   

                                  saledate  
0  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  
1  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  
2  Thu Jan 15 2015 04:30:00 GMT-0800 (PST)  
3  Thu Jan 29 2015 04:30:00 GMT-0800 (PST)  
4  Thu Dec 18 2014 12:30:00 GMT-0800 (PST)

Ajustaría la estratégia de marketing y campañas publicitarias insentivando al uso de los autos usados para una menor inversión en un auto casi nuevo, es decir una reducción de costos iniciales al nuevo cliente, una contribución al medio ambiente, además me enfocaría en la calidad y en la confiabilidad que las marcas le pueden ofrecer a los vehículos, como un respaldo que tendrían los nuevos clientes al elegir el vehículo.

## 5. Agrupación por Marca: Precio y Volumen

a) Agrupa el dataset por `make` y calcula:
   - El promedio de `sellingprice`.
   - La cantidad de ventas por marca (usa el conteo de registros como proxy).

b) Caso de negocio: Si una marca tiene alto volumen pero precios bajos, ¿qué medidas recomendarías para mejorar el margen?

**Paso a paso:**
1. Realiza la agrupación.
2. Renombra las columnas para que sean claras.
3. Interpreta el resultado antes de pasar al siguiente filtro o análisis.

In [ ]:
data.groupby('make')[["sellingprice"]].mean()

sellingprice
make                      
Acura         14017.268260
Aston Martin  54812.000000
Audi          19915.432782
BMW           21441.895748
Bentley       74367.672414
...                    ...
subaru         3710.416667
suzuki         4810.000000
toyota         7339.105263
volkswagen     6145.833333
vw            13672.916667

[96 rows x 1 columns]

In [ ]:
data.groupby('make').agg(precio_promedio=("sellingprice", "mean"),cantidad_ventas=("make", "count")).reset_index().sort_values(by="cantidad_ventas", ascending=False).head()

make  precio_promedio  cantidad_ventas
14       Ford     14540.469648            93554
7   Chevrolet     11977.397528            60197
36     Nissan     11739.015960            53946
49     Toyota     12249.321813            39871
10      Dodge     11206.802592            30710

Cuando la marca tiene un alto volumen pero precios bajos se podría implementar la estratégia de aumento de los precios de los vehículos de manera sutil, con mejoras y la venta de accesorios únicos para darle una mejora en la nueva inversión de los consumidores y generar un atractivo en la compra de los mismos.

## 6. Análisis de Rentabilidad por Marca

a) Crea una nueva columna `profit` que sea la diferencia entre `mmr` y `sellingprice`.
b) Agrupa por `make` y calcula el beneficio promedio para cada marca.

c) Caso de negocio: Identifica las 5 marcas con mayor beneficio promedio y explica qué acciones estratégicas implementarías (por ejemplo, renegociar precios o invertir en calidad).

In [ ]:
data['profit'] = data['mmr'] - data['sellingprice']

In [ ]:
data.groupby('make')[["profit"]].mean().sort_values(by="profit", ascending=False).head()

profit
make                 
Fisker    2638.888889
Ferrari   2000.000000
Plymouth  1577.500000
Bentley   1560.775862
porsche   1419.736842

In [ ]:
data.groupby('make').agg(beneficio_promedio=("profit", "mean")).reset_index().sort_values(by="beneficio_promedio", ascending=False).head()

make  beneficio_promedio
13    Fisker         2638.888889
12   Ferrari         2000.000000
38  Plymouth         1577.500000
4    Bentley         1560.775862
89   porsche         1419.736842

Las acciones estratégicas que implementaría sería implementar actividades VIP para los miembros y nuevos compradores de estas marcas, mostrando siempre la exclusividad de la marca y mejorando la experiencia general de la marca desde el primer momento, esto con el fin de tener lealtad por parte de los clientes.

## 7. Análisis Temporal: Ventas del Año 2015

a) Filtra el dataset para obtener únicamente las ventas del año 2015 y muestra las primeras 10 filas. Cuantos Vehiculos tiene esta dataset


b) Caso de negocio: Si 2015 muestra comportamientos atípicos en comparación con otros años, ¿deberías ajustar la estrategia basándote en tendencias actuales? Explica tu criterio.

In [ ]:
data[data["year"]== 2015].head(10)

year       make    model   trim   body transmission                vin  \
0   2015        Kia  Sorento     LX    SUV    automatic  5xyktca69fg566472   
1   2015        Kia  Sorento     LX    SUV    automatic  5xyktca69fg561319   
3   2015      Volvo      S60     T5  Sedan    automatic  yv1612tb4f1310987   
5   2015     Nissan   Altima  2.5 S  Sedan    automatic  1n4al3ap1fn326013   
11  2015        Kia   Optima     LX  Sedan    automatic  5xxgm4a73fg353538   
12  2015       Ford   Fusion     SE  Sedan    automatic  3fa6p0hdxfr145753   
13  2015        Kia  Sorento     LX    SUV    automatic  5xyktca66fg561407   
15  2015     Nissan   Altima  2.5 S  Sedan    automatic  1n4al3ap5fc124223   
16  2015    Hyundai   Sonata     SE  Sedan    automatic  5npe24af4fh001562   
20  2015  Chevrolet   Impala    LTZ  Sedan    automatic  2g1165s30f9103921   

   state  condition  odometer   color interior  \
0     ca        5.0   16639.0   white    black   
1     ca        5.0    9393.0   white    beige   
3     ca       41.0   14282.0   white    black   
5     ca        1.0    5554.0    gray    black   
11    ca       48.0    2034.0     red      tan   
12    ca        2.0    5559.0   white    beige   
13    ca        5.0   14634.0  silver    black   
15    ca        2.0   11398.0   black    black   
16    ca        NaN    8311.0     red        —   
20    ca       19.0   14538.0  silver    black   

                                               seller      mmr  sellingprice  \
0                             kia motors america  inc  20500.0       21500.0   
1                             kia motors america  inc  20800.0       21500.0   
3                             volvo na rep/world omni  27500.0       27750.0   
5   enterprise vehicle exchange / tra / rental / t...  15350.0       10900.0   
11                                 kia motors finance  15150.0       17700.0   
12  enterprise vehicle exchange / tra / rental / t...  15350.0       12000.0   
13                            kia motors america  inc  20600.0       21500.0   
15  enterprise vehicle exchange / tra / rental / t...  14750.0       14100.0   
16                                           avis tra  15200.0        4200.0   
20  enterprise vehicle exchange / tra / rental / t...  24300.0        7200.0   

                                   saledate   profit  
0   Tue Dec 16 2014 12:30:00 GMT-0800 (PST)  -1000.0  
1   Tue Dec 16 2014 12:30:00 GMT-0800 (PST)   -700.0  
3   Thu Jan 29 2015 04:30:00 GMT-0800 (PST)   -250.0  
5   Tue Dec 30 2014 12:00:00 GMT-0800 (PST)   4450.0  
11  Tue Dec 16 2014 12:00:00 GMT-0800 (PST)  -2550.0  
12  Tue Jan 13 2015 12:00:00 GMT-0800 (PST)   3350.0  
13  Tue Dec 16 2014 12:30:00 GMT-0800 (PST)   -900.0  
15  Tue Dec 23 2014 12:00:00 GMT-0800 (PST)    650.0  
16  Tue Dec 16 2014 13:00:00 GMT-0800 (PST)  11000.0  
20  Tue Jul 07 2015 09:30:00 GMT-0700 (PDT)  17100.0

In [ ]:
ventas_2015=data[data["year"]== 2015]

In [ ]:
ventas_2015.shape

(9437, 17)

Sí, captaría las tendencias actuales del mercado para ajustar los precios y así evitar la depreciación de los autos, a su vez, implementaría estrategias de marketing destacando el valor agregado de los autos.

## 8. Comparación por Tipo de Carrocería

a) Agrupa por la columna `body` y calcula el promedio de `sellingprice` para cada tipo.

b) Caso de negocio: Determina qué tipo de carrocería tiene mayor precio promedio y argumenta cómo podrías aprovechar esta información para posicionar una línea premium o ajustar el inventario.

In [ ]:
data.groupby("body").agg(sellingprice_promedio=("sellingprice", "mean")).reset_index().sort_values(by="sellingprice_promedio", ascending=False).head()

body  sellingprice_promedio
66  granturismo convertible           79041.666667
23  GranTurismo Convertible           74000.000000
5               CTS-V Wagon           50500.000000
55              cts-v coupe           39707.142857
4               CTS-V Coupe           38425.750000

esta información podría aprovecharla con la exposición de carrocería de lujo, ya que por medio de esto podría crear una línea exclusiva para estos autos.

## 9. Filtros Combinados para un Caso Específico

a) **Paso 1:** Agrupa los datos por `state`, `model` y calcula el promedio de `sellingprice` para ver cuál es el precio promedio por combinación.

b) **Paso 2:** Utiliza el resultado del agrupamiento para filtrar el dataset original. Por ejemplo, filtra los registros de un estado ('ca') y un modelo ('Sorento') cuyo `sellingprice` supere el precio promedio calculado.

c) Caso de negocio: Analiza este segmento y plantea qué acciones implementarías para potenciar las ventas en California, considerando el modelo y el segmento premium.

In [ ]:
data.groupby(["state", "model"]).agg(sellingprice_promedio=("sellingprice", "mean")).reset_index().sort_values(by="sellingprice_promedio", ascending=False).head()

state       model  sellingprice_promedio
1966    fl  458 Italia          183000.000000
2679    fl          i8          162000.000000
2240    fl       Ghost          160333.333333
2486    fl  SLS AMG GT          156500.000000
1465    ca          i8          154250.000000

In [ ]:
data[(data["state"]== "ca")& (data["model"]== "i8") & (data["sellingprice"]>162000.000000)]

year make model  trim   body transmission                vin state  \
146917  2014  BMW    i8  Base  Coupe    automatic  wby2z2c59evx64214    ca   

        condition  odometer  color interior                    seller  \
146917        5.0      60.0  white    brown  ferrari of orange county   

             mmr  sellingprice                                 saledate  \
146917  176000.0      165000.0  Thu Jan 15 2015 04:00:00 GMT-0800 (PST)   

         profit  
146917  11000.0

Implementaría estrategias para potenciar las ventas como el ofrecimiento de productos exclusivos de la marca, algunos mantenimientos gratuitos y verificación del estado del vehículo para la captación de nuevos mercados y la atracción por la seguridad del estado del vehículo.

## 10. Ventas Totales por Estado

a) Agrupa por `state` y calcula la suma de `sellingprice` para cada estado.

b) Caso de negocio: Identifica el estado con mayores ventas totales y discute cómo enfocar campañas de marketing o ampliar la red de distribución en ese estado.

In [ ]:
data.groupby("state").agg(sellingprice_total=("sellingprice", "sum")).reset_index().sort_values(by="sellingprice_total", ascending=False).head()

state  sellingprice_total
31    fl        1.151065e+09
29    ca        1.061078e+09
54    pa        8.612346e+08
59    tx        6.064954e+08
32    ga        4.489274e+08

Se pueden fortaleer las alianzas con los concesionarios y varias plataformas digitales para mejorar y agilizar el proceso de compra.

## 11. Rentabilidad por Estado

a) Agrupa por `state` y calcula el promedio de eficiencia operativa. Como no contamos con 'Operating Margin', usa como proxy `profit/sellingprice`.

b) Caso de negocio: Identifica el estado con mayor eficiencia operativa y sugiere estrategias para replicar ese éxito en otros mercados.

In [ ]:
data["efficiency"] = data["profit"] / data["sellingprice"]

In [ ]:
data.groupby("state").agg( eficiencia_promedio=("efficiency", "mean")).reset_index().sort_values(by="eficiencia_promedio", ascending=False).head()

state  eficiencia_promedio
46    nm             1.178797
59    tx             0.565770
33    hi             0.429630
49    ny             0.320112
38    md             0.292650

el estado con mayor eficiencia es nm con un promedio del 1.17, esto quiere decur que generó una ganancia mayor al precio de venta, y para poderlo replicar invertiria en estrategias de marketing para aumentar las ventas de estos vehículos.

## 12. Análisis por Tipo de Transmisión (en lugar de Sales Method)

a) Agrupa el dataset por `transmission` y calcula:
   - La suma de `sellingprice` para cada tipo.
   - El promedio de `sellingprice` para cada tipo.

b) Caso de negocio: ¿Qué tipo de transmisión (automática vs. manual) genera mayores ingresos y qué implicaciones tiene para la estrategia comercial?

In [ ]:
data.groupby("transmission").agg(sellingprice_total=("sellingprice", "sum"),sellingprice_promedio=("sellingprice", "mean")).reset_index().sort_values(by="sellingprice_total", ascending=False).head()

transmission  sellingprice_total  sellingprice_promedio
1    automatic        6.443934e+09           13540.406866
2       manual        1.966936e+08           11211.448358
3        sedan        2.047500e+05           13650.000000
0        Sedan        1.505500e+05           13686.363636

la transmisión automática genera 6.4439.. millones en ventas, y su precio promedio es de 13.540 dólares, esto indica que la estrategia debe ser más promocionada y de esta manera poder ampliar más la oferta de vehículos automáticos, destacando las características más relevantes de los autos.

## 13. Segmentación por Producto Líder

a) Identifica el producto (utiliza la columna `model` como proxy) con mayor cantidad de unidades vendidas. Agrupa por `model` y usa el conteo de registros.

b) Caso de negocio: Si un producto lidera en ventas, ¿cómo ajustarías el inventario y las promociones para maximizar las ganancias?

In [ ]:
data.groupby("model").agg(cantidad_ventas=("model", "count")).reset_index().sort_values(by="cantidad_ventas", ascending=False).head()

model  cantidad_ventas
76   Altima            19349
271   F-150            14479
302  Fusion            12946
151   Camry            12545
257  Escape            11861

lo ajustaría de tal manera pueda tener mi suministro constante asegurado y encontrar proveedores con los que pueda negociar con mejores precios.

## 14. Comparación Crítica entre Retailers

a) Compara dos retailers usando la columna `seller` en cuanto a volumen de ventas y margen de beneficio (usa `profit` como proxy).

b) Caso de negocio: Si un retailer muestra mayores unidades vendidas pero un margen inferior, ¿qué estrategia recomendarías (por ejemplo, optimización de costos, cambio en política de descuentos, etc.)? Justifica tu propuesta.

**Paso a paso:**
1. Agrupa por `seller` y calcula la suma de `sellingprice` y el promedio de `profit`.
2. Ordena los resultados para identificar los líderes en volumen y analiza el margen.
3. Propón una estrategia basada en los resultados.

In [ ]:
data.groupby("seller").agg(volumen_ventas=("seller", "count"),margen_beneficio=("profit", "mean")).reset_index().sort_values(by="volumen_ventas", ascending=False).head()

seller  volumen_ventas  margen_beneficio
9293              nissan-infiniti lt           19693        -70.225969
5262   ford motor credit company llc           19162        216.393122
12747          the hertz corporation           18299        189.344063
11675             santander consumer           15285        380.931305
1536                avis corporation           12540        -20.590510

En este caso se revisaría los costos que tiene la empresa con el fin de optimizar los precios y los costos para poder generar utilidades y beneficios a corto y mediano plazo.

## 15. Caso Integrador: Estrategia Comercial Global

a) **Paso 1:** Identifica cada criterio basado en los análisis anteriores:
   - **Retailer con mayores ventas totales:** Usa la columna `seller` del resumen de retailers.
   - **Estado con mayores ventas totales:** Del agrupamiento por `state`.
   - **Producto con mayor unidades vendidas:** Usa la columna `model`.
   - **Tipo de transmisión con mayores ventas totales:** Agrupa por `transmission`.

b) **Paso 2:** Filtra el dataset original usando cada uno de estos criterios, uno a uno, y luego combínalos en un filtro final.

c) Con base en el análisis combinado, redacta un breve informe (3-4 líneas) proponiendo una estrategia comercial integral para maximizar la rentabilidad, considerando inversión en canales, promociones específicas y optimización de inventario.

d) Caso de negocio: Utiliza la integración de estos criterios para definir en qué segmento enfocar la estrategia global.

In [ ]:
data.groupby("state").agg(ventas_totales=("sellingprice", "sum")).sort_values(by="ventas_totales", ascending=False).head()

ventas_totales
state                
fl       1.151065e+09
ca       1.061078e+09
pa       8.612346e+08
tx       6.064954e+08
ga       4.489274e+08